### ISSUE TITLE
Flight Paths not working

### ISSUE THREAD

**Issue by dipdill** [TrinityCore/TrinityCore#17778]
**Description**:  Flight Paths not working

**Current behaviour**: You can click on a Gryphon Master and try to fly to a known flight location... and it never mounts you to fly there.

**Expected behaviour**:  Mount on a flight path to go to the selected location

**Steps to reproduce the problem**:

Click on a Gryphon Master and select a flight location.

**Branch(es)**: 6.x

**TC hash/commit**:  e6ddb9f2220b

**TDB version**:  TDB 6.04

**Operating system**:  

Server:  Ubuntu 16.04.1 LTS
Client OS: Windows 10 and MacOS 10.11.6
Client Version:  7.0.3.22423

**Comment by tkrokli** [comment]
TrinityCore rev. 0bd6a5d4e5d9 2016-08-14 01:38:25 +0200 (6.x branch) (Win64, Release, Static)
TDB 6.04 + updates up to and including `2016_08_13_00_world.sql`

Confirmed. The same goes for Horde Wind Rider Masters or Bat Handlers as well. You will not be mounted on a taxi to any place you are allowed to fly to when clicking that destination.

The error functionality works, though. So you will see error messages like **You are already there!** (in red) if you select your starting point, **You don't have enough money!** (in yellow) if you want to fly a costly distance or **You haven't reached that flight location on foot yet!** (also in yellow) if you select a high-level area like Hyjal or similar.

Not sure if this has got anything to do with missing SAI / scripting or if it has got something to do with packet handling. Maybe it can be debugged by reading through the Worldserver console error output?

**Comment by lorac2k14** [comment]
Confirmed Flight masters worked up to the change to 7.0.3

**Comment by tkrokli** [comment]
The opcodes change when Blizzard changes the game client by releasing new content, so blame them.
Don't know yet when that opcode will be sniffed and enabled again. Hope it won't take too long.

**Comment by lorac2k14** [comment]
isn't 0x34AA the opcode?

**Comment by tkrokli** [comment]
That is the old one that worked before patch 7.0.3. That is why we get `Received not handled opcode` .

**Comment by lorac2k14** [comment]
To fix go to line 131 in opcode.cpp
and change 
`DEFINE_HANDLER(CMSG_ACTIVATE_TAXI,                                      STATUS_UNHANDLED,  PROCESS_THREADSAFE,   &WorldSession::HandleActivateTaxiOpcode);`
to
`DEFINE_HANDLER(CMSG_ACTIVATE_TAXI,                                      STATUS_LOGGEDIN,  PROCESS_THREADSAFE,   &WorldSession::HandleActivateTaxiOpcode);

I just did this on my side and it works as it should ;)`

**Comment by tkrokli** [comment]
OK, I will test it. About time I started learning more about the opcodes. :+1:

**Comment by lorac2k14** [comment]
I may open a pr with it to get it pushed to source since its a simple fix lol

**Comment by tkrokli** [comment]
I thought something like that should be edited in 2 files, but that is maybe only first implementation.

**Comment by lorac2k14** [comment]
In some cases you have to edit the smsg too but in this case only the cmsg needs to be enabled

**Comment by JaredNinja** [comment]
Experienced this today. Fix seems to be known. Pull requested. Could someone flip the switch? :-D I's love this to be part of the main branch

**Comment by Aokromes** [comment]
pull request is incomplete.

**Comment by JaredNinja** [comment]
Is that something I could help with? Aka:
1. Are we talking about a technical problem?
2. Are we talking about formalities that need to be addressed?

**Comment by Aokromes** [comment]
https://github.com/TrinityCore/TrinityCore/pull/17781#r74732997

**Comment by JaredNinja** [comment]
@Aokromes Yes, I read that. I too like it when people do their homework, and I can appreciate your short and to the point approach too actually.

The thing is, I see big green checkmarks that say its okay for this code to go into main. I also see a discussion about "fields". Don't get me wrong, I am a developer and I have a little bit of knowledge. This project (Trinity) is just new to me, so please bare with me as I am trying to familiarize myself a bit with the inner workings.

Bottom line; I see a great solution by @lorac2k14 , to an annoying problem that is still in the main branch. I cannot judge if the solution is right, you guys can. If it is not going into main because of a little work or paperwork, I'm willing to pick up on that. If you could just give me a little more detail.

Thank you for your time @Aokromes, it's great to see an active member like you on this wonderful project.

**Comment by Aokromes** [comment]
No, it's not a great solution, is a incomplete solution based on incomplete research, that's why it's unmerged. Also, you missed the most important thing:
Aokromes added the Feedback-FixHack label on Aug 25

**Comment by JaredNinja** [comment]
@Aokromes Understood. That we don't want.

Is anyone that you know of working on this problem? If not, it might be a great exercise for me to get involved.

**Comment by tkrokli** [comment]
Get sniffs of the exact opcodes from retail to see what it should be now, instead of the old ones.

**Comment by tkrokli** [comment]
BTW, it would not hurt if someone took the time to fix some of the opcodes during login. They still add a bit of noise to the error logs in TC 6.x, but I'm not proficient with opcodes & related code.
